scripts/docker/genesis.Dockerfile

```diff
@@ -0,0 +1,20 @@
+# syntax=docker/dockerfile:experimental
+
+# Usage:
+#   DOCKER_BUILDKIT=1 docker build \
+#      -t concordium/genesis:<tag> \
+#      --build-arg=tag=<genesis-data-repo-tag> \
+#      --build-arg=dir_path=<genesis-data-repo-path> \
+#      -f genesis.Dockerfile \
+#      --ssh=default \
+#      .
+
+FROM alpine/git:latest as data
+RUN mkdir -p -m 0600 ~/.ssh && ssh-keyscan gitlab.com >> ~/.ssh/known_hosts
+ARG tag
+ARG dir_path
+RUN --mount=type=ssh git clone --depth=1 --branch="${tag}" git@gitlab.com:Concordium/genesis-data.git /tmp/genesis-data && \
+    mv "/tmp/genesis-data/${dir_path}" /genesis-data && rm -rf /tmp/genesis-data
```

Comment:
I think this is a really bad idea. genesis data contains sensitive information and doing this makes it very easy to use the wrong path and leak testnet keys.

For testnet and mainnet we have public genesis blocks which can be used.

Reply:
I also think it is a misuse of a docker container to use it as a zip archive.

Reply:
This is a temporary container that exists only during build time. If you look below, only `genesis.dat` is copied into the final image. The other files never leave the build process. Even with caching, they are deleted in the same command as they are cloned and therefore don't get into the layer.

Also these images don't get published publicly, only in the internal ECR repo.

What makes it a misuse? I think it's a pretty common use case for init containers to inject data into the "main" ones.

Reply:
Can we proceed on this?

Reply:
In any case, my main objection is to the cloning of the repository to get a file out. Even though it is only during the build it is copying keys all over which is not good practice.

I'm OK if you want to have `genesis:mainnet` and `genesis:testnet` images with publicly accessible testnet and mainnet genesis blocks.

Reply:
I disagree that it's error prone - there is literally no parameters you can give this job to make it leak the keys.

But what I wrote about caching is not entirely correct. In the current form, the keys could get into the cache, although only on a machine that has the credentials to get them anyway.

Anyway, after the genesis location has been parameterized, this has become more of a convenience thing that we can live without. So I will remove it. If we need it after all, the genesis-data repo may also be a better place to put it.